packages/api/src/external/carequality/process-outbound-patient-discovery-resps.ts

```diff
@@ -102,6 +136,129 @@ function buildCQLinks(pdResults: OutboundPatientDiscoveryResp[]): CQLink[] {
+}): Promise<boolean> {
+  const updatedPatient = await getPatientOrFail(patient);
+
+  const newDemographicsHere = getNewDemographics(updatedPatient, cqLinks);
+  const foundNewDemographicsHere = newDemographicsHere.length > 0;
+  const foundNewDemographicsAcrossHies = await checkLinkDemographicsAcrossHies({
+    patient: updatedPatient,
+    requestId,
+  });
+  if (foundNewDemographicsHere || foundNewDemographicsAcrossHies) {
+    if (foundNewDemographicsHere) {
+      await updatePatientLinkDemographics({
+        requestId,
+        patient: updatedPatient,
+        source: MedicalDataSource.CAREQUALITY,
+        links: newDemographicsHere,
+      });
+    }
+    await discover({
+      patient: updatedPatient,
+      facilityId,
+      rerunPdOnNewDemographics: false,
+    });
+    analytics({
+      distinctId: updatedPatient.cxId,
+      event: EventTypes.patientDiscovery,
+      properties: {
+        hie: MedicalDataSource.CAREQUALITY,
+        patientId: updatedPatient.id,
+        requestId,
+        foundNewDemographicsHere,
+        foundNewDemographicsAcrossHies,
+      },
+    });
+    return true;
+  }
+  return false;
+}
+
+export async function runNexPdIfScheduled({
+  patient,
+  requestId,
+}: {
+  patient: Patient;
+  requestId: string;
+}): Promise<boolean> {
+  const updatedPatient = await getPatientOrFail(patient);
+
+  const scheduledPdRequest = getCQData(updatedPatient.data.externalData)?.scheduledPdRequest;
+
+  if (scheduledPdRequest) {
+    await discover({
+      patient: updatedPatient,
+      facilityId: scheduledPdRequest.facilityId,
+      requestId: scheduledPdRequest.requestId,
+      forceEnabled: scheduledPdRequest.forceCarequality,
+      rerunPdOnNewDemographics: scheduledPdRequest.rerunPdOnNewDemographics,
+    });
+    await resetPatientScheduledPatientDiscoveryRequestId({
+      patient: updatedPatient,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+    analytics({
+      distinctId: updatedPatient.cxId,
+      event: EventTypes.runScheduledPatientDiscovery,
+      properties: {
+        hie: MedicalDataSource.CAREQUALITY,
+        patientId: updatedPatient.id,
+        requestId,
+        scheduledPdRequestId: scheduledPdRequest.requestId,
+      },
+    });
+    return true;
+  }
+  return false;
+}
+
+export async function queryDocsIfScheduled({
+  patient,
+  isFailed = false,
+}: {
+  patient: Pick<Patient, "id" | "cxId">;
+  isFailed?: boolean;
+}): Promise<void> {
+  const updatedPatient = await getPatientOrFail(patient);
+
+  const scheduledDocQueryRequestId = getCQData(
+    updatedPatient.data.externalData
+  )?.scheduledDocQueryRequestId;
+
+  if (scheduledDocQueryRequestId) {
+    if (isFailed) {
+      await setDocQueryProgress({
+        patient: updatedPatient,
+        requestId: scheduledDocQueryRequestId,
+        source: MedicalDataSource.CAREQUALITY,
+        downloadProgress: { status: "failed", total: 0 },
+        convertProgress: { status: "failed", total: 0 },
```

Comment:
I don't think we should set those as `failed`. From my understanding of the code so far, what failed was PD, not download or conversion, right?

Reply:
@leite08 this is how we do it now. I did align CW with CQ in this regard (both set it this way on failure).

https://github.com/metriport/metriport/blob/6b14fc71ac69ce3a25b73933b84865f5fd8dec9f/packages/api/src/external/carequality/process-patient-discovery-progress.ts#L37

Reply:
I see, and now I understand. It's because this is being run only if there was a scheduled DQ, so we mark that as failed.

Couple thoughts:
1. This requires remembering to understand what it means, like "Why did DQ failed?" and then remember that it can fail if it was scheduled and the PD failed - so the reason is not that DQ really failed. Should we add a `statusReason` so we can store a string to helps us understand what's going on w/ the system?
1. We don't schedule conversion, so I still think we shouldn't set `convertProgress` to failed. What if we just reset it?

Reply:
Decision from call: leave as is (i.e. in line with what we do now).

https://metriport.slack.com/archives/C04DMKE9DME/p1717549824389949